File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int i = 0;
        double sum = 0;
        double maxAvg = -1e9;
        for (int j = 0; j < nums.size(); j++) {
            sum += nums[j];
            if (j - i + 1 == k) {
                maxAvg = max(maxAvg, sum / k);
                sum -= nums[i];
                i++;
            }
        }
        return maxAvg;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {
    let i = 0;
    let sum = 0
    let maxSum = -Infinity;
    for (let j = 0; j < nums.length; j++) {
        sum += nums[j]
        if (j - i + 1 == k) {
            maxSum = Math.max(maxSum, sum)
            sum -= nums[i]
            i++
        }

    }

    return maxSum / k

};